honeybee/aperture.py

```diff
@@ -0,0 +1,506 @@
+                should be flipped. Default is False to have contours on top or right.
+                Setting to True will start contours on the bottom or left.
+            indoor: Boolean for whether louvers should be generated facing the
+                opposite direction of the aperture normal (typically meaning
+                indoor geometry). Default: False.
+
+        Returns:
+            louvers: A list of Shade objects that cover the Aperture.
+        """
+        angle = math.radians(angle)
+        louvers = []
+        ap_geo = self.geometry if indoor is False else self.geometry.flip()
+        shade_faces = ap_geo.countour_fins_by_distance_between(
+            distance, depth, offset, angle, contour_vector, flip_start_side)
+        for i, shade_geo in enumerate(shade_faces):
+            louvers.append(Shade('{}_Shd{}'.format(self.name_original, i), shade_geo))
+        return louvers
+
+    def move(self, moving_vec):
+        """Move this Aperture along a vector.
+
+        Args:
+            moving_vec: A ladybug_geometry Vector3D with the direction and distance
+                to move the face.
+        """
+        self._geometry = self.geometry.move(moving_vec)
+
+    def rotate(self, axis, angle, origin):
+        """Rotate this Aperture by a certain angle around an axis and origin.
+
+        Args:
+            axis: A ladybug_geometry Vector3D axis representing the axis of rotation.
+            angle: An angle for rotation in degrees.
+            origin: A ladybug_geometry Point3D for the origin around which the
+                object will be rotated.
+        """
+        self._geometry = self.geometry.rotate(axis, math.radians(angle), origin)
+
+    def rotate_xy(self, angle, origin):
+        """Rotate this Aperture counterclockwise in the world XY plane by a certain angle.
+
+        Args:
+            angle: An angle in degrees.
+            origin: A ladybug_geometry Point3D for the origin around which the
+                object will be rotated.
+        """
+        self._geometry = self.geometry.rotate_xy(math.radians(angle), origin)
+
+    def reflect(self, plane):
+        """Reflect this Aperture across a plane with the input normal vector and origin.
+
+        Args:
+            plane: A ladybug_geometry Plane across which the object will
+                be reflected.
+        """
+        self._geometry = self.geometry.reflect(plane.n, plane.o)
+
+    def scale(self, factor, origin=None):
+        """Scale this Aperture by a factor from an origin point.
+
+        Args:
+            factor: A number representing how much the object should be scaled.
+            origin: A ladybug_geometry Point3D representing the origin from which
+                to scale. If None, it will be scaled from the World origin (0, 0, 0).
+        """
+        self._geometry = self.geometry.scale(factor, origin)
+
+    def is_planar(self, tolerance, raise_exception=True):
```

Comment:
```suggestion
    def is_planar(self, tolerance, raise_exception=False):
```

This might be how I would use such method but by default you want get a True / False unless there is a case for checking planarity as a requirement which is different.

Reply:
Do you want me to set `raise_exception` to `False` for all ~20 of the checks that I implemented throughout the library?  Or Just these 2?  I feel that we should probably be consistent across the library but I don't have a strong opinion about whether they should all be `False` or all be `True`. I just know that I'm going to forget if these 2 are the only exceptions.

Reply:
I mean it for all. Basically I think when one uses `face.is_planar()` it should not throw an exception by default. It should return True or False.

Reply:
Ok. Good to know.  I remember you saying a while back that we should use the practice of always setting `raise_exception` to `True` for functions that check things but I see your opinion has changed here.  I will use a practice of setting it to `False` here in `honeybee_core` and I'll consider using the same practice elsewhere across the libraries. Or, if you think defaulting to `False` is not good practice for other libraries, let me know if there are any reasons why `honeybee-core` is the exception to the rule.

Reply:
I have probably mis-communicated what I meant. These are two different cases in my mind.

Case 1 is a check for making a follow up decision and case 2 is the check as a requirement. `is_planar` to me is more of a case 1. If it was `check_planarity` then I would have expected it to raise an exception is it is not planar.

Reply:
I'm sorry to say that I am even more confused now.  So the decision about how the method is named should determine whether `raise_exception` is `True` or `False`?  So maybe we should be changing the name of the method to 'check_planarity` rather than the `raise_exception` default?
I think that I'll just leave it the way it is now since I don't think that you have gotten the chance to look at the other ~20 checks methods in this PR.  After you have taken a look at those, then let me know what you think here.